src/Microsoft.Identity.Web/TokenAcquisition.cs

```diff
@@ -347,6 +351,19 @@ private async Task<IConfidentialClientApplication> BuildConfidentialClientApplic
             string authority;
             IConfidentialClientApplication app;
 
+            MicrosoftIdentityOptionsValidation microsoftIdentityOptionsValidation = new MicrosoftIdentityOptionsValidation();
```

Comment:
You could also inject a `IValidateOptions<MicrosoftIdentityOptions>` in the constructor by dependency injection

Reply:
I tried that, but I only need to check the secret here, not all the options, so i just need the class and that method. it didn't work passing it into the constructor.